neo/SmartContract/InteropService.cs

```diff
@@ -201,6 +202,21 @@ private static bool Runtime_Serialize(ApplicationEngine engine)
             return true;
         }
 
+        private static bool Runtime_GetNotifications(ApplicationEngine engine)
+        {
+            var data = engine.CurrentContext.EvaluationStack.Pop().GetByteArray();
+            if (data.Length != UInt160.Length) return false;
+
+            var hash = new UInt160(data);
+            engine.CurrentContext.EvaluationStack.Push
+                (
+                new VM.Types.Array(engine.Notifications
```

Comment:
You should check the array size. Also, should we consider using `Enumerator`?

Reply:
What do you prefer, array or enumerator? i think that it could be easier if is an array?

Reply:
Array is very limited, in size and resources...Enum gives us hope to filter data quickly, and use it efficiently even over array max limit.
A System.Enumerator.Filter would be nice too, the challenge is to pass a lambda inside it. I used to defend OPEVAL for that, but perhaps we can do something safer and simpler.